### ISSUE TITLE
TAG Ambassadors

### ISSUE THREAD

**Issue by leonardpahlke** [cncf/toc#1211]
Hey, I'd like to explore the idea of linking CNCF Ambassadors with CNCF TAGs, anticipating a potential synergistic impact within the community.

* **[CNCF Ambassadors](https://www.cncf.io/people/ambassadors/)**: This community group comprises passionate individuals involved in diverse capacities within the cloud native community. 
* **[CNCF Technical Advisory Groups](https://github.com/cncf/toc/tree/main/tags#current-tags)**: TAGs extend the TOC in specific domains, fostering domain maturity within the cloud native community and projects.

### Potential

* The current Ambassadors program effectively empowers individuals in their existing cloud native contributions, offering perks and a platform for connecting with like-minded folks. The program, especially the 2.0 revamp, is a big success. However, as an Ambassador, you may ask yourself, how does this role help me to become a more impactful contributor. This addition might help with that. Aligning Ambassadors with TAGs could provide a structured framework for their contributions beyond the contributions they do already (+ potentially diving into other areas).
* While TAGs follow slightly different governance models, they all interact with projects and maintainers, necessitating effective communication of their work. Some Ambassadors are project maintainers, while others are already extensively involved in communication efforts (Podcasts, YouTube, Social Media, Blogs, etc.). Given that Ambassadors are typically vocal and engaged community members, supporting TAGs in this capacity seems like a natural fit. Ambassadors could act as project liaisons or in an outreach role, or anything the TAG seems fit.

### Comments

TAGs would need to define an Ambassador role within their respective governance structures. It's important to note that a TAG Ambassador role does not necessarily need to be exclusive to CNCF Ambassadors (it could even be an option to qualify yourself as CNCF Ambassador).

cc @k8tgreenley (CNCF Ambassador program)

**Comment by csantanapr** [comment]
As a CNCF Ambassador and Project Maintainer I like this initiative +1

**Comment by TheFoxAtWork** [comment]
+1 Thank you Leo! I love this idea. CNCF Ambassadors have a unique connection into the ecosystem that can provide insights and opportunities for TAGs to explore as well as increase exposure of TAGs to support both adopters and projects where they're at.

**Comment by linsun** [comment]
+1 love the idea as well, as a project maintainer and also a CNCF ambassador. Would love to have ambassadors participate and contribute to related TAG meetings and related activity to increase the impact of the TAG(s).

**Comment by alolita** [comment]
Good suggestion @leonardpahlke In the Observability TAG, some of the CNCF ambassadors focused on observability projects participate in the TAG discussions, publishing posts and encouraging projects to present in the TAG meetings. We could formalize the relationship further.

**Comment by OrlinVasilev** [comment]
+1 love the idea

**Comment by scottrigby** [comment]
hey @leonardpahlke yes! We discussed this. Support 100%. I'd also like to see closer ties between CNCF projects and TAGs. This combined with your idea above would also lead to more active ties between ambassadors and projects.

+1 from me

**Comment by angellk** [comment]
+1 fantastic idea

**Comment by raravena80** [comment]
I have a suggestion for an action plan (open to changes/removals/additions, or other suggestions):
1. Create a github project under the TOC repo (or another CNCF repo) (create one called something like TAGs-Ambassadors)
1. Ask TAG chairs/leads to add items to the project where they think they may need help from ambassadors.  Label these issues something like help-wanted or help-needed. 
1. After some time (say 2 weeks), ask TAG chairs/leads to review the complete list of issues and check which ones would be applicable to their respective TAGs.  For the ones that are, ask them to label the issue with `tag-<TAG name>`. 
   1. For the issues that have been labeled with `tag-<TAG name>`, ask chairs/leads to open a matching CNCF TAG repo github issue and reference it back to the TOC or top level issue.  TAGs can use projects or issues to track these within their repos (I think it would be up to each individual TAG).
   1. Also give TAG chairs/leads a chance to add any new items if they'd like.
1. Once the help-wanted or help-needed issues have been established we (anyone or any group in the TOC/TAGs/community really) can reach out to ambassadors to volunteer if they are interested in tackling a specific issue.
   1. The issue could be worked on within a single or multiple TAGs.  It's open to the Ambassador's choice, wrt the amount of work that they'd like to take on.   
1. After an Ambassador has agreed to help with an issue, TAG chairs/leads/members can follow up with the ambassador to check on progress.
1. At any point in time community members, including TOC/TAG chairs/leads can create issues in the top level github repo and tie it to their TAG repos.
   1. Ambassadors can also take on issues at anytime.
1. After some time (say 4-6 months) iterate back to 2.
   1. This would possibly help re-establish traction.

**Comment by amye** [comment]
@raravena80 - could it just be here in this repo? We add a view to the project board and let people work in that?

**Comment by raravena80** [comment]
> @raravena80 - could it just be here in this repo? We add a view to the project board and let people work in that?

Yep. I think that would work. But it would be good to hear any other feedback too.

**Comment by leonardpahlke** [comment]
Giving the response to this issue, the ones I got in person in Chicago, and that this comes from both fellow TAG Leads and fellow CNCF Ambassadors, I think this idea is worth following up 👍.

Thank you so much @TheFoxAtWork for your input. Good questions to kick off further discussions! To pick this up, I would now open a docs file to answer these questions and perhaps a few more to get towards a detailed proposal. I will ask in the next days over Slack for folks to contribute.

@raravena80, that's another approach to connect Ambassadors with the TAGs. This reminds me of https://clotributor.dev/, a tool WIP to browse issues. However, I like the idea of integrating Ambassadors in TAGs so they can play a more active part and continue building a team, extending the TAG teams. Expanding the TAGs' governance roles (see https://github.com/cncf/toc/pull/1116) by adding a TAG Ambassador role, as proposed in this issue, seems to be for me the obvious way to go about it.

**Comment by leonardpahlke** [comment]
> I believe the action plan that I shared would be helpful in making CNCF Ambassadors becoming TAG Ambassadors in a similar way that TAG contributors become chairs/leads

Ah, gotcha! 👍 It's a proposal for defining an onboarding program for ambassadors that are interested becoming a TAG Ambassador. They can browse a project board etc.

**Comment by TheFoxAtWork** [comment]
Okay so next steps:
- recommend inviting a few CNCF ambassadors to each TAG's meeting to identify areas they think they can have an impact
- collaboration between TAG leadership and CNCF Ambassadors to develop a proposal, engagement, and set expectations on interactions
- submit PR and present to the TOC at an upcoming meeting (date TBD subject to existing schedule presenter(s) availability) so we can formalize.

**Comment by leonardpahlke** [comment]
Yes! I created this doc to work on a proposal. Any help with this would be very much appreciated. 

Steps to move forward:
- [x] Created an initial proposal draft structure to enable collaboration 
- [ ] Shared the initial proposal with TAG Chairs and CNCF Ambassadors + CNCF personnel 
- [ ] Started to write the proposal
- [ ] Discussed the proposal in a public TOC meeting
- [ ] Refined and iterated (TOC meeting comments)
- [ ] Share the proposal once again 2 week notice
- [ ] Refined and iterated
- [ ] Submit the proposal to the TOC

DOC: https://docs.google.com/document/d/1ISSzLcdQbGZBz6z8noF1qrRH80AxDXNCICYZzrGqPuM/edit?usp=sharing

**Comment by leonardpahlke** [comment]
The next TAG ENV meeting is scheduled for the 21st of Feb. I will invite a couple of Ambassadors to the meeting to discuss how they would see the TAG Ambassador role. @raravena80 perhaps you could do the same for TAG Runtime?

**Comment by raravena80** [comment]
> The next TAG ENV meeting is scheduled for the 21st of Feb. I will invite a couple of Ambassadors to the meeting to discuss how they would see the TAG Ambassador role. @raravena80 perhaps you could do the same for TAG Runtime?

FYI. We are discussing on slack the 'how' to reach out to Ambassadors.

**Comment by PushkarJ** [comment]
Just wanted to use this opportunity to call out that we have already have following super amazing ambassadors contributing to TAG Security:

- https://www.cncf.io/people/ambassadors/?p=annalisa-gennaro led the Italian translation of cloud native security whitepaper
- https://www.cncf.io/people/ambassadors/?p=carolina-valencia led the Spanish translation of the same paper 
- https://www.cncf.io/people/ambassadors/?p=jon-zeolla is running the controls catalog project and is a prolific contributor to our TAG since many years
- https://www.cncf.io/people/ambassadors/?p=savitha-raghunathan a contributor to v2 of the same white paper above 

I am sure I missed a few but yes agree 💯 - ambassadors make the TAGs awesome! I encourage and welcome any other ambassadors interested in contributing to Security TAG too to reach out to us on #tag-security channel :)

**Comment by lizrice** [comment]
I want to mention a related thought that has come up a couple of times at governing board meetings, which is that CNCF Ambassadors could have a role liaising directly with projects and their maintainers, helping projects to make the most out of being in the CNCF. Ambassadors could for example be checking in regularly with projects to see how things are going, perhaps going to regular project meetings, helping projects man their booths at KubeCon, and so on. This could help build stronger relationships and better two-way communication between CNCF and its projects.

**Comment by leonardpahlke** [comment]
@jberkus - The details of the TAG Ambassador role are right now subject to discussion, so this is not set in stone. First of all, i think, the manner in which TAGs may incorporate Ambassadors is likely to vary across different TAGs. Therefore, I am thinking to propose a variety of ways the TAG Ambassador role could be integrated within TAGs. In the [proposal draft document](https://docs.google.com/document/d/1ISSzLcdQbGZBz6z8noF1qrRH80AxDXNCICYZzrGqPuM/edit?usp=sharing) (which is still a work in progress), currently suggest two primary methods for a TAG Ambassador to undertake their role within a TAG. Ultimately, it will be up to the TAG Chairs to determine how these roles could be beneficial within the “guidelines” defined by the proposal. This approach is very much WIP and open to comments based on your and others' feedback.

1. Ambassadors as project or open source foundation liaisons
     1. TAG Ambassador serving as a liaison to CNCF projects
     2. TAG Ambassador serving as a liaison to open source foundations
2. Ambassadors as community organizers
     1. TAG Ambassador organizing community events

These tasks, especially the first one, are usually done by the chairs, right. But in my experience in TAG ENV, the help is very important to get better at it. TAG Chairs have a number of tasks and this sometimes falls a bit short (IMO), but is very important. In some cases, Chairs or TLs are inherently connected to projects or other foundations, in which case the role of TAG Ambassador for that project/foundation may be redundant. For other organizations and projects, establishing a TAG Ambassador can be very helpful. Someone who stays in contact with both groups and communicates between them. + the second part which points towards a role where TAG Ambassador supports the TAG at events, leads the organization of person meetups, etc., perhaps brings maintainers to the TAGs to write blogs etc. (we will see, not defined in detail yet) 

—

> (@lizrice) [...] Ambassadors could for example be checking in regularly with projects to see how things are going, perhaps going to regular project meetings, helping projects man their booths at KubeCon, and so on. [...]

👍, I will add this to the proposal, thanks!

**Comment by windsonsea** [comment]
+1 support this initiative

**Comment by angellk** [comment]
Idea: Technical Community Groups - encourage Ambassadors to be community organizers for the groups.

**Comment by krol3** [comment]
Hi @leonardpahlke , This seems like an excellent idea, especially with the technical community group, where ambassadors can help lead it. How can I support you, help you?

**Comment by leonardpahlke** [comment]
Thanks for the reminder @krol3 -- happy to pick this up again this fall. More than happy to collab with you and anyone else on this and land the idea with the Ambassadors && TAGs. @krol3 if you find the time feel free to comment the linked docs file. 🙌

**Comment by mrbobbytables** [comment]
looping in Audra (will find github handle later)
